it/src/test/java/com/google/cloud/teleport/it/spanner/DefaultSpannerResourceManagerTest.java

```diff
@@ -0,0 +1,312 @@
+            + ") PRIMARY KEY (SingerId)";
+
+    // act & assert
+    assertThrows(SpannerResourceManagerException.class, () -> testManager.createTable(statement));
+  }
+
+  @Test
+  public void testCreateTable_throwsExceptionWhenSpannerCreateDatabaseFails()
+      throws ExecutionException, InterruptedException {
+    // arrange
+    when(spanner.getDatabaseAdminClient().createDatabase(any(), any(), any()).get())
+        .thenThrow(InterruptedException.class);
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act & assert
+    assertThrows(SpannerResourceManagerException.class, () -> testManager.createTable(statement));
+  }
+
+  @Test
+  public void testCreateTable_throwsExceptionWhenSpannerUpdateDatabaseFails()
+      throws ExecutionException, InterruptedException {
+    // arrange
+    when(spanner.getDatabaseAdminClient().updateDatabaseDdl(any(), any(), any(), any()).get())
+        .thenThrow(InterruptedException.class);
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act & assert
+    assertThrows(SpannerResourceManagerException.class, () -> testManager.createTable(statement));
+  }
+
+  @Test
+  public void testCreateTable_worksWhenSpannerDoesntThrowAnyError() {
+    //   arrange
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act
+    testManager.createTable(statement);
+
+    // assert
+    verify(spanner.getDatabaseAdminClient())
+        .updateDatabaseDdl(
+            argThat(s -> s.startsWith("test-id") && !s.equals("test-id")),
+            eq("test_id"),
+            eq(ImmutableList.of(statement)),
+            any());
+  }
+
+  @Test
+  public void testWriteSingleRecord_worksWhenSpannerWriteSucceeds() {
+    // arrange
+    when(spanner.getDatabaseClient(any()).write(any())).thenReturn(Timestamp.now());
+    testManager.createTable("");
+    Mutation testMutation =
+        Mutation.newInsertOrUpdateBuilder("SingerId")
+            .set("SingerId")
+            .to(1)
+            .set("FirstName")
+            .to("Marc")
+            .set("LastName")
+            .to("Richards")
+            .build();
+
+    // act
+    testManager.write(testMutation);
+
+    // assert
+    verify(spanner.getDatabaseClient(any())).write(writeMutationCaptor.capture());
+    Iterable<Mutation> actualWriteMutation = writeMutationCaptor.getValue();
+    assertThat(actualWriteMutation).isEqualTo(ImmutableList.of(testMutation));
```

Comment:
For iterables, it's best to use `containsExactly` or `containsExactlyElementsIn`: https://javadoc.io/doc/com.google.truth/truth/latest/com/google/common/truth/IterableSubject.html

Some iterables won't play nice with `isEqualTo`, so if anything changes about the underlying implementation, this may start failing.